Lib/Src/error_log.cpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file error_log.cpp
+ */
+
+/*
+ *  Created on: Feb 12, 2022
+ *      Author: niuslar
+ */
+
+#include "error_log.h"
+
+/**
+ * @brief Constructor
+ * @param Pointer to CUartCom object
+ * @param module_name Identifier that will be added to the message
+ */
+CLog::CLog(CUartCom *error_uart, const std::string module_name)
+    : mp_error_uart(error_uart),
+      m_module_name(module_name)
+{
+}
+
+/**
+ * @brief change log level to change what is sent
+ * @param log_level:
+ * 			LOG_ERROR = will show only errors
+ * 			LOG_WARNING = will show error and warnings
+ * 			LOG_INFO = will show everything
+ */
+void CLog::setLogLevel(log_level_t log_level)
+{
+    m_log_level = log_level;
+}
+
+/**
+ * @brief method to log errors, warnings and info
+ * @param log_level Level of the message (LOG_ERROR, LOG_WARNING, LOG_INFO)
+ * @param message
+ */
+void CLog::log(log_level_t log_level, const std::string &message)
+{
+    if (log_level == LOG_INFO && m_log_level >= LOG_INFO)
+    {
+        // Convert to C style string
+        const char *msg = ((const std::string)("[INFO]->" + m_module_name +
```

Comment:
Where is the *msg pointing? Who owns that memory? When `mp_erorr_uart->sendMessage(msg) is executed, does it copy the message or is it going to just use the pointer as if memory is safe?

Reply:
Thanks, I realised c_str() wasn't very reliable. I changed this method to use C++ strings instead and moved the conversion to the CUartCom sendMessage method.